examples/drawing/src/undo-redo.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {useEffect, useReducer, useRef} from 'react';
+import {ControlPosition, MapControl, useMap} from '@vis.gl/react-google-maps';
+
+import reducer from './reducer';
+
+import {
+  DrawResult,
+  DrawingActionKind,
+  Overlay,
+  isCircle,
+  isMarker,
+  isPolygon,
+  isPolyline,
+  isRectangle
+} from './types';
+
+interface Props {
+  drawingManager: google.maps.drawing.DrawingManager | null;
+}
+
+export const UndoRedo = ({drawingManager}: Props) => {
```

Comment:
Could this be renamed to `UndoRedoControl`? 

Another related Idea would be to extract the effects from here into reducer.ts (and rename that to undo-redo.ts), so there's one place where both data definition and manipulation happen. This file would then be only the UI parts of Undo/Redo.